docs/HowTo/develop-gds-plugins.md

```diff
@@ -0,0 +1,164 @@
+        return "my-plugin"
+
+    @classmethod
+    def get_arguments(cls):
+        """ Arguments to request from the CLI """
+        return {}
+
+    @classmethod
+    def check_arguments(cls):
+        """ Check arguments from the CLI """
+        pass
+
+    @classmethod
+    @gds_plugin_implementation
+    def register_framing_plugin(cls):
+        """ Register the MyPlugin plugin """
+        return cls
+```
+
+### Implementing Virtual Functions
+
+Each plugin virtual-class (e.g. `FramerDeframer`) has a set of virtual methods that plugin implementors *must* implement
+in order to support the plugin implementation.  These functions are marked as ` @abc.abstractmethod` and can be found in
+the virtual class definition.
+
+[FramerDeframer virtual functions](https://github.com/fprime-community/fprime-gds/blob/devel/src/fprime_gds/common/communication/framing.py#L29-L51)
+consist of a `frame` and `deframe` method. This guide will use no-op functions.
+
+**`src/my_plugin.py`:**
+```python
+from fprime_gds.common.communication.framing import FramerDeframer
+from fprime_gds.plugin.definitions import gds_plugin_implementation
+
+class MyPlugin(FramerDeframer):
+
+    def frame(self, data):
+        """ Frames data by no-op """
+        return b"" + data # Forces a copy
+
+    def deframe(self, data, no_copy=False):
+        """ Deframe data by no-op """
+        if no_copy:
+            return data
+        return b"" + data # Forces a copy
+        
+    @classmethod
+    def get_name(cls):
+        """ Name of this implementation provided to CLI """
+        return "my-plugin"
+
+    @classmethod
+    def get_arguments(cls):
+        """ Arguments to request from the CLI """
+        return {}
+
+    @classmethod
+    def check_arguments(cls):
+        """ Check arguments from the CLI """
+        pass
+
+    @classmethod
+    @gds_plugin_implementation
+    def register_framing_plugin(cls):
+        """ Register the MyPlugin plugin """
+        return cls
+```
+
+This is the basic implementation of a no-argument framing plugin.  Next, this guide will cover how to integrate this
+plugin via python packaging.  Then arguments will be covered.
+
+### Distributing Plugins
+
+Plugins are supplied as python packages with an entrypoint used to load the plugin. In the root of the package a basic
+python package will need to be configured. This consists of two files: pyproject.toml representing the package, and
+setup.py for backwards compatibility.
+
+The project structure should look like:
+```
+    src/my_plugin.py
+    pyproject.toml
+    setup.py
+```
+
+A sample pyproject.toml file would look like:
+
+```toml
+[build-system]
+requires = ["setuptools", "setuptools_scm[toml]>=6.2", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "fprime-gds-my-plugin"
+dynamic = ["version"]
+dependencies = [
+    "pluggy>=1.3.0",
+    "fprime-gds>=3.4.4"
+]
+
+[project.entry_points."fprime_gds"]
+my_plugin = "my_plugin:MyPlugin"
```

Comment:
I'm not familiar with pluggy - are these two lines what makes it that the `fprime-gds` will be able to pick up the plugin even though it's installed in a different package, because they are sharing entrypoints?

Reply:
Yes.  It follows the form:

```
[project.entry_points."name_of_project_loading_plugin"]
unique_name_for_plugin = "module_path_to_plugin:class_name_defining_plugin"
```